server/src/main/java/org/elasticsearch/index/query/PinnedQueryBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
```

Comment:
Can we move this class to the client module ? It is misleading to have it here and the duplication would be the same in the client module. We already made the choice to duplicate the request/response for a lot of APIs so let's not change the model for queries or aggs.

Reply:
I thinks that's what I'd like @hub-cap to confirm - exactly what package in client module?
There is no precedent at the moment for QueryBuilder classes. Unlike request/response classes all of the existing QueryBuilder classes used by HLRC live in `server` module and my assumption was that PinnedQueryBuilder should be there too and be migrated along with all the other existing QueryBuilder classes when the time comes to decide how HLRC will create queries without a dependency on `server`.

Reply:
I still need a decision on this.
Summarising the various positions stated so far:

Module|Why
-----|-----
`server` | Historic reason - that's where all other QueryBuilders live and from where clientside classes will be generated in future. I _think_ this is @hub-cap's [preference](https://github.com/elastic/dev/issues/1244#issuecomment-519490192)
`client` module | That's where HLRC client code _should_ belong (@jimczi's [preference](https://github.com/elastic/elasticsearch/pull/44345#discussion_r312834428) stated above)
`xpack` | That's where non-OSS functions should exist (@nknize's  [suggestion](https://github.com/elastic/dev/issues/1244#issuecomment-519538337)).